diff -ru qtbase_/src/corelib/serialization/qxmlstream.cpp qtbase/src/corelib/serialization/qxmlstream.cpp
--- qtbase_/src/corelib/serialization/qxmlstream.cpp	2022-02-16 21:13:47.000000000 +0800
+++ qtbase/src/corelib/serialization/qxmlstream.cpp	2023-07-09 12:14:13.301671600 +0800
@@ -1292,7 +1292,7 @@
     return n;
 }
 
-inline int QXmlStreamReaderPrivate::fastScanName(int *prefix)
+inline int QXmlStreamReaderPrivate::fastScanName(Value *val)
 {
     int n = 0;
     uint c;
@@ -1329,16 +1329,16 @@
         case '+':
         case '*':
             putChar(c);
-            if (prefix && *prefix == n+1) {
-                *prefix = 0;
+            if (val && val->prefix == n + 1) {
+                val->prefix = 0;
                 putChar(':');
                 --n;
             }
             return n;
         case ':':
-            if (prefix) {
-                if (*prefix == 0) {
-                    *prefix = n+2;
+            if (val) {
+                if (val->prefix == 0) {
+                    val->prefix = n + 2;
                 } else { // only one colon allowed according to the namespace spec.
                     putChar(c);
                     return n;
@@ -1354,8 +1354,8 @@
         }
     }
 
-    if (prefix)
-        *prefix = 0;
+    if (val)
+        val->prefix = 0;
     int pos = textBuffer.size() - n;
     putString(textBuffer, pos);
     textBuffer.resize(pos);
diff -ru qtbase_/src/corelib/serialization/qxmlstream.g qtbase/src/corelib/serialization/qxmlstream.g
--- qtbase_/src/corelib/serialization/qxmlstream.g	2022-02-16 21:13:47.000000000 +0800
+++ qtbase/src/corelib/serialization/qxmlstream.g	2023-07-09 12:11:03.215930700 +0800
@@ -1488,7 +1488,8 @@
 qname ::= LETTER;
 /.
         case $rule_number: {
-            sym(1).len += fastScanName(&sym(1).prefix);
+            Value &val = sym(1);
+            val.len += fastScanName(&val);
             if (atEnd) {
                 resume($rule_number);
                 return false;
diff -ru qtbase_/src/corelib/serialization/qxmlstream_p.h qtbase/src/corelib/serialization/qxmlstream_p.h
--- qtbase_/src/corelib/serialization/qxmlstream_p.h	2022-02-16 21:13:47.000000000 +0800
+++ qtbase/src/corelib/serialization/qxmlstream_p.h	2023-07-09 12:11:18.187661000 +0800
@@ -503,7 +503,7 @@
     int fastScanLiteralContent();
     int fastScanSpace();
     int fastScanContentCharList();
-    int fastScanName(int *prefix = nullptr);
+    int fastScanName(Value *val = nullptr);
     inline int fastScanNMTOKEN();
 
 
diff -ru qtbase_/src/corelib/serialization/qxmlstreamparser_p.h qtbase/src/corelib/serialization/qxmlstreamparser_p.h
--- qtbase_/src/corelib/serialization/qxmlstreamparser_p.h	2022-02-16 21:13:47.000000000 +0800
+++ qtbase/src/corelib/serialization/qxmlstreamparser_p.h	2023-07-09 12:12:15.837780800 +0800
@@ -982,7 +982,8 @@
         break;
 
         case 262: {
-            sym(1).len += fastScanName(&sym(1).prefix);
+            Value &val = sym(1);
+            val.len += fastScanName(&val);
             if (atEnd) {
                 resume(262);
                 return false;
