diff -rNu a/src/corelib/global/qlogging.cpp b/src/corelib/global/qlogging.cpp
--- a/src/corelib/global/qlogging.cpp	2023-10-17 21:45:08.000000000 +0800
+++ b/src/corelib/global/qlogging.cpp	2024-03-13 15:22:40.794680400 +0800
@@ -108,6 +108,7 @@
 
 #ifndef QT_BOOTSTRAPPED
 #if QT_CONFIG(regularexpression)
+# ifndef Q_OS_ANDROID
 #  ifdef __UCLIBC__
 #    if __UCLIBC_HAS_BACKTRACE__
 #      define QLOGGING_HAVE_BACKTRACE
@@ -115,6 +116,7 @@
 #  elif (defined(__GLIBC__) && defined(__GLIBCXX__)) || (__has_include(<cxxabi.h>) && __has_include(<execinfo.h>))
 #    define QLOGGING_HAVE_BACKTRACE
 #  endif
+# endif
 #endif
 
 #if defined(Q_OS_LINUX) && (defined(__GLIBC__) || __has_include(<sys/syscall.h>))
