diff -rNu a/src/corelib/global/qlogging.cpp b/src/corelib/global/qlogging.cpp
--- a/src/corelib/global/qlogging.cpp	2024-01-05 03:21:43.000000000 +0800
+++ b/src/corelib/global/qlogging.cpp	2024-03-12 08:11:13.641763000 +0800
@@ -102,6 +102,7 @@
 #endif
 
 #if QT_CONFIG(regularexpression)
+# ifndef Q_OS_ANDROID
 #  ifdef __UCLIBC__
 #    if __UCLIBC_HAS_BACKTRACE__
 #      define QLOGGING_HAVE_BACKTRACE
@@ -109,6 +110,7 @@
 #  elif (defined(__GLIBC__) && defined(__GLIBCXX__)) || (__has_include(<cxxabi.h>) && __has_include(<execinfo.h>))
 #    define QLOGGING_HAVE_BACKTRACE
 #  endif
+# endif
 #endif
 
 #if QT_CONFIG(slog2)
