diff -rNu a/src/plugins/avfoundation/camera/avfcamerautility.mm b/src/plugins/avfoundation/camera/avfcamerautility.mm
--- a/src/plugins/avfoundation/camera/avfcamerautility.mm	2024-05-19 17:33:18
+++ b/src/plugins/avfoundation/camera/avfcamerautility.mm	2024-05-19 17:28:08
@@ -101,9 +101,9 @@
     }
 };
 
-struct FormatHasNoFPSRange : std::unary_function<AVCaptureDeviceFormat *, bool>
+struct FormatHasNoFPSRange
 {
-    bool operator() (AVCaptureDeviceFormat *format)
+    bool operator() (AVCaptureDeviceFormat *format) const
     {
         Q_ASSERT(format);
         return !format.videoSupportedFrameRateRanges || !format.videoSupportedFrameRateRanges.count;
