diff -rNu a/src/3rdparty/assimp/src/code/AssetLib/FBX/FBXBinaryTokenizer.cpp b/src/3rdparty/assimp/src/code/AssetLib/FBX/FBXBinaryTokenizer.cpp
--- a/src/3rdparty/assimp/src/code/AssetLib/FBX/FBXBinaryTokenizer.cpp	2023-09-24 19:38:48.000000000 +0800
+++ b/src/3rdparty/assimp/src/code/AssetLib/FBX/FBXBinaryTokenizer.cpp	2024-03-13 15:19:12.264815300 +0800
@@ -50,7 +50,7 @@
 #include "FBXTokenizer.h"
 #include "FBXUtil.h"
 #include <assimp/defs.h>
-#include <stdint.h>
+#include <cstdint>
 #include <assimp/Exceptional.h>
 #include <assimp/ByteSwapper.h>
 #include <assimp/DefaultLogger.hpp>
